contracts/ConvictionVoting.sol

```diff
@@ -73,11 +80,14 @@ contract ConvictionVoting is AragonApp, TokenManagerHook {
     mapping(address => uint256[]) internal voterStakedProposals;
 
     event ConvictionSettingsChanged(uint256 decay, uint256 maxRatio, uint256 weight, uint256 minThresholdStakePercentage);
-    event ProposalAdded(address indexed entity, uint256 indexed id, string title, bytes link, uint256 amount, address beneficiary);
+    event ProposalAdded(address indexed entity, uint256 indexed id, uint256 indexed actionId, string title, bytes link, uint256 amount, address beneficiary);
     event StakeAdded(address indexed entity, uint256 indexed id, uint256  amount, uint256 tokensStaked, uint256 totalTokensStaked, uint256 conviction);
     event StakeWithdrawn(address entity, uint256 indexed id, uint256 amount, uint256 tokensStaked, uint256 totalTokensStaked, uint256 conviction);
     event ProposalExecuted(uint256 indexed id, uint256 conviction);
-    event ProposalCancelled(uint256 indexed id);
+    event ProposalPaused(uint256 indexed proposalId, uint256 indexed actionId);
+    event ProposalResumed(uint256 indexed proposalId, uint256 indexed actionId);
+    event ProposalCancelled(uint256 indexed proposalId, uint256 indexed actionId);
```

Comment:
Wondering if we should keep the cancelled action separated, like one cancel for when the user remove the proposal from consideration and the other cancel when the disputable action is rejected

Reply:
Do you think it's likely we will differentiate between them on the UI? Only I ask because at some point I'm going to need to optimize this contract, as it's already over the byte size limit, and will likely need to remove some stuff. However, if it would be useful then I'll work it out.

Reply:
mmhh yeah i think that at some point in the ui we might need to show something if the proposal was cancelled by the disputable flow cc @lkngtn @fioreb. If we don't want to add another event we might want to just add a flag on the cancel event or something like that

Reply:
If we end up needing the differentiation, we could use the proposal previous status to know the reason it got cancelled.

Reply:
I'll add it, I think it could be useful.

Reply:
Oh yes, I need to work on the disputable flow for conviction voting, but I think when is cancelled by the disputable flow it will look similar to how it looks in the admin votes.